plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoTableHandle.java

```diff
@@ -33,12 +34,13 @@
     private final SchemaTableName schemaTableName;
     private final RemoteTableName remoteTableName;
     private final TupleDomain<ColumnHandle> constraint;
+    private final Optional<Set<MongoColumnHandle>> projectedColumns;
```

Comment:
Can we change to non-optional `Set` type instead?

Reply:
No. The empty Set is a different information then Optional.empty(). If I have a query like `SELECT COUNT(42 + 42) from orders` (actual test that we have in a BCT) it will go into an infinite loop with the `PushProjectionIntoTableScan` rule. We have the same logic in the BigQuery connector.

Reply:
That should be possible. We already do this in Iceberg connector. No change requested though.

Reply:
Tried. It worked. Please review now.